_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+     
+     $$y_{n+1} -2 y_{n-1} + y_{n-2} = h f_{n}$$
+     
+     The characteristic polynomial is given by
+     
+     $$\rho(z) = z^3 -2z -1$$
+     
+     One of the roots of this polynomial satisfies \(|z|>1\) and this implies that the method is not zero-stable.
+     </p>
+     <p>
+     Exercise 12.19:
+     The first equation can be found by substituting \(f(t,y) = \lambda y\) in equation (12.51) in the book and by solving for \(k_1,k_2\) (it is a \(2\times 2\) linear system). Substituting the values of \(A\) and \(b\) from the Butcher tableau in this formula and in the one right before equation (12.51) in the book, and simplifying, we get the formula for \(R(\lambda h)\). Finally, \(p\) and \(q\) are given by \(p,q=-3\pm i \sqrt{3}\). One can see that this implies \(|R(z)|<1\) if \(Re(z) <0\) and thus the method is A-stable.
+     </p>
+     </details>
+
+<br />
+
+# 3 ResNets
+  **Motivation**: The introduction of Residual Networks (ResNets) made possible to train very deep networks. In this section we study some residual architectures variants and their properties. We then look into how ResNets approximates ODEs and how this interpretation can motivate neural net architectures and new training approaches.  This is important in order to understand the basic models underlying Neural ODEs and gain some insights on their connection to numerical solutions of ODEs.
+
+  **Topics**:
+
+  1. ResNets
+  2. ResNets and ODEs
+
+  **Required Reading**:
+
+  1. ResNets: 
+     * [ResNets](https://www.coursera.org/lecture/convolutional-neural-networks/resnets-HAhz9) 
+     * [An Overview of ResNet and its Variants](https://towardsdatascience.com/an-overview-of-resnet-and-its-variants-5281e2f56035)
+  2. ResNets and ODEs: 
+     * Sections 1-3 from [Multi-level Residual Networks from Dynamical Systems View](https://arxiv.org/pdf/1710.10348.pdf)
+     * [Reversible Architectures for Arbitrarily Deep Residual Neural Networks](https://arxiv.org/abs/1709.03698)
+     * Invertible ResNets: [The Reversible Residual Network: Backpropagation Without Storing Activations](https://arxiv.org/pdf/1707.04585.pdf)
+     * [Stable Architectures for Deep Neural Networks](https://arxiv.org/pdf/1705.03341.pdf)
+     
+  **Notes**: Here is a [link](/assets/nodes_notes/week3.pdf) to our notes for the lesson.
+  
+  **Optional Reading**:
+
+  1. The original ResNets paper: [Deep Residual Learning for Image Recognition](https://arxiv.org/abs/1512.03385)
+  2. Another blog post on ResNets: [Understanding and Implementing Architectures of ResNet and ResNeXt for state-of-the-art Image Classification](https://medium.com/@14prakash/understanding-and-implementing-architectures-of-resnet-and-resnext-for-state-of-the-art-image-cf51669e1624)
+  
+  **Questions**:
+
+  1. Do you understand why adding ‘residual layers’ should not degrade the network performance?
+     <details><summary>Solution</summary>
+     <p>
+     See the notes from this week.
```

Comment:
This isn't the best. Why not just duplicate / expand it here?

Reply:
added

Reply:
The latex could be a lot better here.

<img width="853" alt="Screen Shot 2019-08-17 at 11 13 34 AM" src="https://user-images.githubusercontent.com/615351/63213927-193ac500-c0e0-11e9-8049-ef518fbe7533.png">

Reply:
fixed it